### DISCUSSION TITLE
How to upgrade conductor from v2.30.1 to v3.3.0

### DISCUSSION THREAD

**Discussion by xiaowei582648206** [Netflix/conductor#2516] [Q&A]
How can I upgrade conductor to **V3.3.0**,  here is my questions:

1. the changes of conductor database table structure is to huge ( remove the primary key of tables), when I upgrade to V3.3.0,   what influence it will be caused?
2. the chagnes of conductor task definition is also significant （ replace the **DECISION** task with SWITCH task ... ）, Is it backward compatible ?
3. could you give us some suggestion about upgrading without effecting on **PRODUCT** environment?

thanks a lot!

**Comment by aravindanr** [comment]
1. The [changelog](https://github.com/Netflix/conductor/blob/main/CHANGELOG.md) lists the changes from 2.31 to 3.x. The database schema did not change between the releases. Can you elaborate "remove the primary key of tables"?
2. The DECISION task is deprecated not removed. Your existing workflows using the DECISION task will work without any issues.
3. We can't offer advice about specific production environment, as each is unique and have different constraints. If you run into issues during the upgrade, we are here to help.

**Reply by xiaowei582648206** [reply]
in `conductor/mysql-persistence/src/main/resources/db/migration/V8__update_pk.sql` , this will drop the old primary key( column :id),add the new primary key， what will be the impact of this change？

![image](https://user-images.githubusercontent.com/5030860/138232396-dde9c90b-e185-4e2b-9658-7a326343fec2.png)
